lib/routes/account.js

```diff
@@ -891,12 +893,37 @@ module.exports = function (
       handler: function (request, reply) {
         log.begin('Account.RecoveryEmailStatus', request)
         var sessionToken = request.auth.credentials
-        reply(
-          {
-            email: sessionToken.email,
-            verified: sessionToken.emailVerified
+
+        cleanUpIfAccountInvalid()
+          .then(createResponse)
+          .done(reply, reply)
+
+        function cleanUpIfAccountInvalid() {
+          // Some historical bugs mean we've allowed creation
+          // of accounts with invalid email addresses. These
+          // can never be verified, so the best we can do is
+          // to delete them so the browser will stop polling.
+          if (!sessionToken.emailVerified) {
+            if (!validators.isValidEmailAddress(sessionToken.email)) {
+              return db.deleteAccount(sessionToken)
+                .then(
+                  function () {
+                    // Act as though we deleted the account asynchronously
+                    // and caused the sessionToken to become invalid.
+                    throw error.invalidToken()
+                  }
```

Comment:
I'm just wondering what the internal browser polling does when it get's this error. Is 401/110 a signal to the browser to clean up the pending account signup?